cmake/Common.cmake

```diff
@@ -21,24 +23,52 @@ function(package_set_postinst_prerm LIB_NAMES LIB_DIRS INCLUDE_DIRS)
         set(POSTINST_SOURCE "${POSTINST_SOURCE}\nln -sr ${lib_dir}/lib${lib_name}.so ${lib_dir}/../../lib/lib${lib_name}.so")
         set(POSTINST_SOURCE "${POSTINST_SOURCE}\nln -sr ${lib_dir}/cmake/${lib_name} ${lib_dir}/../../lib/cmake/${lib_name}\n")
 
+        set(PREINST_SOURCE "${PREINST_SOURCE}\nif [ $1 == 2 ]; then")
+        set(PREINST_SOURCE "${PREINST_SOURCE}\n\trm /etc/ld.so.conf.d/${lib_name}.conf")
+        set(PREINST_SOURCE "${PREINST_SOURCE}\n\tunlink ${inc_dir}/../../include/${lib_name}")
+        set(PREINST_SOURCE "${PREINST_SOURCE}\n\tunlink ${lib_dir}/../../lib/lib${lib_name}.so")
+        set(PREINST_SOURCE "${PREINST_SOURCE}\n\tunlink ${lib_dir}/../../lib/cmake/${lib_name}/${lib_name}")
+        set(PREINST_SOURCE "${PREINST_SOURCE}\n\trm ${lib_dir}/lib${lib_name}.so*")
+        set(PREINST_SOURCE "${PREINST_SOURCE}\n\trm -d ${lib_dir}/../../lib/cmake/${lib_name}\n")
+        set(PREINST_SOURCE "${PREINST_SOURCE}\nfi")
+
         set(PRERM_SOURCE "${PRERM_SOURCE}\nrm /etc/ld.so.conf.d/${lib_name}.conf")
         set(PRERM_SOURCE "${PRERM_SOURCE}\nunlink ${inc_dir}/../../include/${lib_name}")
         set(PRERM_SOURCE "${PRERM_SOURCE}\nunlink ${lib_dir}/../../lib/lib${lib_name}.so")
         set(PRERM_SOURCE "${PRERM_SOURCE}\nunlink ${lib_dir}/../../lib/cmake/${lib_name}/${lib_name}")
         set(PRERM_SOURCE "${PRERM_SOURCE}\nrm -d ${lib_dir}/../../lib/cmake/${lib_name}\n")
+
+        set(PRERM_RPM_SOURCE "${PRERM_RPM_SOURCE}\nif [ $1 == 0 ]; then")
+        set(PRERM_RPM_SOURCE "${PRERM_RPM_SOURCE}\n\trm /etc/ld.so.conf.d/${lib_name}.conf")
+        set(PRERM_RPM_SOURCE "${PRERM_RPM_SOURCE}\n\tunlink ${inc_dir}/../../include/${lib_name}")
+        set(PRERM_RPM_SOURCE "${PRERM_RPM_SOURCE}\n\tunlink ${lib_dir}/../../lib/lib${lib_name}.so")
+        set(PRERM_RPM_SOURCE "${PRERM_RPM_SOURCE}\n\tunlink ${lib_dir}/../../lib/cmake/${lib_name}/${lib_name}")
+        set(PRERM_RPM_SOURCE "${PRERM_RPM_SOURCE}\n\trm -d ${lib_dir}/../../lib/cmake/${lib_name}\n")
+        set(PRERM_RPM_SOURCE "${PRERM_RPM_SOURCE}\nfi")
     endforeach()
     file(WRITE ${PROJECT_BINARY_DIR}/deb/postinst ""
         "#!/bin/bash\n"
         "${POSTINST_SOURCE}\n"
         "ldconfig\n"
     )
+    file(WRITE ${PROJECT_BINARY_DIR}/deb/preinst ""
```

Comment:
Can a single comment be added indicating when the 4 scripts will be called by the installer.